docs/src/Typeahead.doc.js

```diff
@@ -0,0 +1,193 @@
+        type: 'string',
+      },
+      {
+        name: 'noResultText',
+        type: 'string',
+        href: 'basicExample',
+        required: false,
+        description: 'The text shown when the input value returns no matches',
+      },
+      {
+        name: 'onBlur',
+        type: '({ event: SyntheticFocusEvent<>, value: string }) => void',
+        required: false,
+        description: 'Callback when you focus outside the component ',
+        href: 'basicExample',
+      },
+
+      {
+        name: 'onChange',
+        type: '({ event: SyntheticInputEvent<>, value: string }) => void',
+        required: false,
+        description: 'Callback when user types into the control input field',
+        href: 'basicExample',
+      },
+
+      {
+        name: 'onFocus',
+        type: '({ event: SyntheticFocusEvent<>, value: string }) => void',
+        required: false,
+        description: 'Callback when you focus on the component',
+        href: 'basicExample',
+      },
+      {
+        name: 'onSelect',
+        type: '({ event: SyntheticInputEvent<>, value: string }) => void',
+        required: false,
+        description: 'Callback when you select an item ',
+        href: 'basicExample',
+      },
+      {
+        name: 'placeholder',
+        type: 'string',
+        href: 'basicExample',
+      },
+      {
+        name: 'searchField',
+        type: 'string',
+        description:
+          'The property name on which you want to be able to search on',
+        href: 'basicExample',
+        defaultValue: 'label',
+      },
+
+      {
+        name: 'size',
+        type: '"md" | "lg"',
+        required: false,
+        description: 'md: 40px, lg: 48px',
+        defaultValue: 'md',
+      },
+    ]}
+  />
+);
+
+card(
+  <Example
+    id="basicExample"
+    name="Example"
+    defaultCode={`
+      function Example(props) {
+        const [item, setItem] = React.useState('');
+        const [selected, setSelected] = React.useState(null);
+
+        const options = Array
+                          .from(Array(100).keys())
+                          .map(item => ({value: "value-" + item, label: "Label-" + item}))
+
+        const handleOnChange = ({value}) => {
+          setItem(value);
+        }
+
+        const handleSelect = item => {
+          setSelected(item)
+        }
+
+        return (
+          <>
+            <Box marginBottom={2}><Text>Selected Item: {selected && selected.label || ""}</Text></Box>
+
+            <Typeahead
+              id="Typeahead-example"
+              noResultText="No Results"
+              data={options}
+              placeholder="Select a Label"
+              onChange={handleOnChange}
+              value={item}
+              onSelect={handleSelect}
+            />
+          </>
+        );
+      }
+    `}
+  />
+);
+
+card(
+  <Example
+    id="defaultItemExample"
+    name="Default Item Example"
+    defaultCode={`
+      function Example(props) {
```

Comment:
@simkessy thanks for adding the up/down arrow key functionality. Could we only update the `selected` option when the user hits enter? Similar as to https://material-ui.com/components/autocomplete/ does it?

Right now, the selected option changes as you go up and down with the arrow keys, which is unexpected
![typeahead-arrow-change](https://user-images.githubusercontent.com/127199/85907481-7e624b00-b7c6-11ea-975d-8ba646042ac0.gif)